component-library/component-lib/src/lib/form-components/multi-checkbox/multi-checkbox.component.ts

```diff
@@ -34,123 +38,223 @@ export class MultiCheckboxComponent implements OnInit {
+                  this.config!.parent!.mixed = true;
+                } else if (positive > 0 && negative == 0) {
+                  this.config.parent?.formGroup
+                    .get(this.config.parent?.id)
+                    ?.patchValue(true, { emitEvent: false });
+
+                  this.config!.parent!.mixed = false;
+                } else if (positive == 0 && negative > 0) {
+                  this.config.parent?.formGroup
+                    .get(this.config.parent?.id)
+                    ?.patchValue(false, { emitEvent: false });
+                  this.config!.parent!.mixed = false;
+                }
+                this.checkError(
+                  this.config?.parent?.formGroup?.get(this.config.parent?.id)
+                    ?.status,
+                  this.config.parent?.formGroup || new FormGroup({}),
+                  this.config.parent?.id || ''
+                );
+
+                this.config.children?.forEach((res) => {
+                  this.checkError(
+                    this.config?.parent?.formGroup?.get(res.id)?.status,
+                    this.config?.parent?.formGroup || new FormGroup({}),
+                    res.id
+                  );
+                });
+              }
+            }
           }
+        );
+    }
+  }
+
+  checkErrorsSubscription() {
+    this.config.children?.forEach((res) => {
+      res.formGroup?.get(res.id)?.statusChanges.subscribe((value: any) => {
+        this.checkError(value, res.formGroup, res.id);
+        value === 'DISABLED' ? this.disabledStatus = true : this.disabledStatus = false;
+      });
+    });
+
+    if (this.config.parent !== undefined) {
+      this.config.parent.formGroup
+        ?.get(this.config.parent.id)
+        ?.statusChanges.subscribe((value: any) => {
+          this.checkError(
+            value,
+            this.config?.parent?.formGroup || new FormGroup({}),
+            this.config?.parent?.id || ''
+          );
+          value === 'DISABLED' ? this.disabledStatus = true : this.disabledStatus = false;
+
+          this.config.children?.forEach((res) => {
+            res.formGroup
+              ?.get(res.id)
+              ?.statusChanges.subscribe((value: any) => {
+                this.checkError(value, res.formGroup, res.id);
+                value === 'DISABLED' ? this.disabledStatus = true : this.disabledStatus = false;
+              });
+          });
+        });
+    }
+  }
+
+  checkError(value: any, group: FormGroup, id: string) {
+    if (value !== 'VALID') {
+      for (const error in group.get(id)?.errors) {
+        let errorIndex = this.config.errorMessages?.findIndex((errorPair) => {
+          return errorPair.key === error;
+        });
+
+        let errorMessagesKeys = this.errorMessages?.findIndex((errorPair) => {
+          return errorPair.key === error && id === errorPair.id;
+        });
+
+        if (errorIndex > -1 && errorMessagesKeys === -1) {
+          this.errorMessages.push({
```

Comment:
Took me a while to figure out what errorMessages is for because it has the same name as the field in the config. Can you maybe rename it?